packages/cli/internal/pkg/cli/workflow/manager.go

```diff
@@ -436,6 +475,23 @@ func (m *Manager) setWorkflowParameters() {
 	m.workflowParams["workflowInputs"] = filepath.Base(m.attachments[0])
 }
 
+func (m *Manager) setWorkflowEngineParameters() {
+	if m.err != nil {
+		return
+	}
+	m.workflowEngineParams = make(map[string]string)
+	if m.optionFileUrl == "" {
+		return
+	}
+	if m.options != nil {
+		if m.workflowEngine == "nextflow" || m.workflowEngine == "miniwdl" {
+			m.err = fmt.Errorf("optionFile flag cannot be used with head node engines")
+			return
+		}
+		m.workflowEngineParams = m.options
```

Comment:
How do we pass this to the actual engine?

Reply:
The parameter is set in `runWorkflow()` with `option.WorkflowEngineParams(m.workflowEngineParams))`